docs/metrics/coefficient-of-determination.md

```diff
@@ -0,0 +1,77 @@
+# Coefficient of Determination (R²)
+
+The Coefficient of Determination, commonly denoted as R², is a statistical measure used to assess the goodness of fit
+of a regression model. It represents the proportion of the variance in the dependent variable that is predictable from
+the independent variables.
+
+R² provides a scale that is **generally** from 0 to 1, where higher values indicate a better fit and imply that
+the model can better explain the variation of the output with the input variables. It is particularly useful for
+comparing the explanatory power of different models and their overall fit. Negative values are possible and indicate
+that the mean of the data itself is a better fit to the data than the regressor/model itself.
+
+## Implementation Details
+
+R² is calculated as the proportion of the total variation of outcomes explained by the model. Mathematically, it can
+be represented as:
+
+$$
+R² = 1 - \frac{\sum_{i=1}^{N}(y_i - \hat{y}_i)^2}{\sum_{i=1}^{N}(y_i - \bar{y})^2}
+$$
+
+where $y_i$ is the actual value, $\hat{y}_i$ is the predicted value from the model, $\bar{y}$ is the mean of the
+actual values, and $N$ is the total number of observations.
+
+### Examples
+
+Temperature Estimation:
+
+<div class="grid" markdown>
+| Ground Truth Temperature (°C) | Predicted Temperature (°C) |
+| ----------------------------- | -------------------------- |
+| 25                            | 27                         |
+| 35                            | 30                         |
+
+$$
+\begin{align}
+R² &= 1 - \frac{(25 - 27)^2 + (35 - 30)^2}{(25 - 30)^2 + (35 - 30)^2} \\
+&= 0.42
+\end{align}
+$$
+</div>
+
+Age Estimation:
+
+<div class="grid" markdown>
+| Ground Truth Age (Years) | Predicted Age (Years) |
+| ------------------------ | --------------------- |
+| 60                       | 70                    |
+| 40                       | 20                    |
+
+$$
+\begin{align}
+R² &= 1 - \frac{(60 - 70)^2 + (40 - 20)^2}{(60 - 50)^2 + (40 - 50)^2} \\
+&= -1.5
+\end{align}
+$$
+</div>
+
+## Limitations and Biases
+
+When assessing regression models on unseen data, R² quantifies how well model predictions align with observed outcomes
+indicating the variance in the dependent variable explained by the model. However, its utility is nuanced:
+
+- R² might overly favor the model's performance if the unseen data's distribution diverges from the training data,
```

Comment:
What does it mean by overly favoring? Will R^2 be large when there is data characteristic shifts?

Reply:
Oh sorry - will try to make it a bit clearer.

Basically if unseen data diverges from training data. (And the test-data is not representative of that shift - the R^2 value might give a bad fit). ~fitting the distribution well but not generalizing well (kind of similar to overfitting actually - so will change the wording),

Reply:
I've changed it but if its not of too much use - am open to removing it as well